packages/grid/x-data-grid/src/utils/fastMemo.ts

```diff
@@ -0,0 +1,6 @@
+import * as React from 'react';
+import { fastObjectShallowCompare } from './fastObjectShallowCompare';
+
+export function fastMemo<T>(component: T): T {
```

Comment:
I think that it could be great to leave a comment that explains how this comparison is different from the one of React https://github.com/facebook/react/blob/4ddc019aca8e08fd59cb43de5e0032be77d6174e/packages/react-reconciler/src/ReactFiberBeginWork.js#L570

Reply:
it goes [faaaaast](https://github.com/mui/mui-x/pull/9037#discussion_r1214288683), but yeah we should explain also `fastObjectShallowCompare`, it's tailored for a specific use-case.